engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2AuthenticationResources.scala

```diff
@@ -36,17 +37,28 @@ class OAuth2AuthenticationResources(realm: String, service: OAuth2Service[Authen
 
   override lazy val additionalRoute: Route =
     pathEnd {
-      parameters('code) { authorizeToken =>
+      parameters('code,  'redirect_uri.?) { (authorizationCode, redirectUri) =>
         get {
-          complete {
-            oAuth2Authenticate(authorizeToken)
+          xor(redirectUri, configuration.redirectUri.map(_.toString)).map { redirectUri =>
```

Comment:
You can just do `redirectUri.orElse(configuration.redirectUri.map(_.toString))`

Reply:
It is supposed to be an eXclusive OR.